### DISCUSSION TITLE
Wix4 managed BA can't be uninstalled in quiet mode

### DISCUSSION THREAD

**Discussion by EvgenPervenenko** [wixtoolset/issues#7966] [Questions] [answered]
Hi, community, I am using custom BA UI wix4, and Apply method from IEngine.cs takes parent window hwnd

`void Apply(IntPtr hwndParent) `

but if run uninstall in quiet mode getting next error:

`Error 0x80070057: BA passed NULL hwndParent to Apply.`

For quiet mode handler of parent window is NULL, because uninstallation run without window, is there a way to do it?

**Comment by robmen** [comment]
> because uninstallation run without window, is there a way to do it?

You need to provide a window so that required Windows messages are processed, but you don't need to show the window. This was a bug fix in WiX v4.

**Reply by EvgenPervenenko** [reply]
@robmen thanks, is this bug fix already released I am using WixToolset.Mba.Core version 4.0.3

**Reply by robmen** [reply, marked as answer]
Sorry, my wording was not great. The window is required in WiX v4 to fix a bug found in v3. The actual issue is linked in the v3 to v4 migration doc.

The window _should_ have been required in v3.

**Reply by EvgenPervenenko** [reply]
@robmen thank you
